src/ssh_fmt_plug.c

```diff
@@ -317,17 +318,19 @@ static void *get_salt(char *ciphertext)
 			OPENSSL_free(nm);
 			OPENSSL_free(header);
 			OPENSSL_free(data);
-			OPENSSL_free(psalt->bp);
 		}
 	}
+	OPENSSL_free(nm);
```

Comment:
@magnumripper  
Yes. I have pay attention to the line 317(318). There is a **for()** loop, it can only broken by `break` (line 310 and 315). When it breaks, **the `nm` is not freed**. So I free the `nm` after the **for()** loop.